### DISCUSSION TITLE
How to add third party scripts inside <head></head> Component while exporting

### DISCUSSION THREAD

**Discussion by fahad-qureshi786** [GrapesJS/grapesjs#4560] [Q&A]
Hi Grapesjs Community,

Good day, I have been struggling with this problems for several days. I developed components like Connect Wallet, and wrote business logic for Metamask wallet connection. When I export the code from my canva, it export business logic very well but I am unable to get the libraries connection between the head section of the html page. Please refer to the screenshot below
![image](https://user-images.githubusercontent.com/75601362/188726152-12e44be4-f3ec-4fad-ae5c-ad58e053ab5e.png)

Code After export, I want the libraries come there,
![image](https://user-images.githubusercontent.com/75601362/188726347-799aeafd-a724-4547-b57c-adfcb1ded675.png)

To understand the problem we added a link to our website. Please visit our deployed product [Web3Makr](https://web3makr.netlify.app/) and drag and drop the Wallet Connection (Image 1) component, export code and you will witness its not being connected with the wallet (See the login() function) , then you do it by yourself by importing following two scripts inside head tag
```
   <script src="https://cdn.jsdelivr.net/npm/web3@latest/dist/web3.min.js"></script>
   <script src="https://unpkg.com/moralis-v1/dist/moralis.js"></script>
```

Then refresh local page, and then see metamask wallet will up and running.

I would appreciate your help. Thanks in advance

**Comment by artf** [comment]
One option would be to inject stuff in the export code by relying on the `root` option from the [export plugin](https://github.com/artf/grapesjs-plugin-export).
Another alternative is to append those scripts dynamically in your [custom component script code](https://grapesjs.com/docs/modules/Components-js.html). In that case, they will be appended only when the component is actually dropped in the canvas.

**Reply by fahad-qureshi786** [reply]
Thanks @artf I will test this thing and will ping here if anything I found working, I appreciate it. 

Thanks

**Reply by fahad-qureshi786** [reply]
@artf 

Hi Sir, This didn't work. Can we have a quick call where I can show you the issue. 

Thanks in advance
